docs/reference/resource-limits-fees.mdx

```diff
@@ -13,26 +13,49 @@ Resource limitations and fees only apply to smart contract transactions. Read mo
-| Ledger entry size (including Wasm entries) per Txn | 2 KB | 64 KB |
-| Read/Write Ledger entries per Txn | 3 read; 2 write | 40 read; 25 write |
-| Read/Write bytes per Txn | 3.2 KB read; 3.2 KB write | 130 KB read; 65 KB write |
-| Transaction size | 10 KB | 70 KB |
-| Persistent entry minimal/initial lifetime | 4,096 ledgers (~5.68 hours) | 120 days |
-| Temporary entry minimal/initial lifetime | 16 ledgers | 1 day |
-| Max ledger entry expiration bump | 61 days | 6 months |
-| Events+return value size bytes | 200 B | 8 KB |
+| Network Setting | Phase 0 | Phase 1 (current) | Phase 2 |
+| --- | --- | --- | --- |
+| Soroban Txn per ledger | 1 | 100 | 125 |
+| Max CPU Instructions per Txn | 2.5 million (2M instructions for max Wasm) | 100 million (VM instantiation consumes up to 23M for a 64KB Wasm, and 46M for multiple Wasms of 130KB) | 500 million |
+| Memory limit per Txn | 2 MB | 40 MB | 42 MB |
+| Ledger entry size (including Wasm entries) per Txn | 2 KB | 64 KB | 66 KB |
+| Read/Write Ledger entries per Txn | 3 read; 2 write | 40 read; 25 write | 40 read; 25 write |
+| Read/Write bytes per Txn | 3.2 KB read; 3.2 KB write | 130 KB read; 65 KB write | 133 KB read; 66 KB write |
+| Transaction size | 10 KB | 70 KB | 70 KB |
+| Persistent entry minimal/initial lifetime | 4,096 ledgers (~5.68 hours) | 120 days | 120 days |
+| Temporary entry minimal/initial lifetime | 16 ledgers | 1 day | 1 day |
+| Max ledger entry expiration bump | 61 days | 6 months | N/A |
+| Events+return value size bytes | 200 B | 8 KB | N/A |
 
 ## Resource Fees
 
 Note that write fees grow linearly from empty ledger to ledger "target size", and then grow linearly, but with a 1000x factor after exceeding the target. This is to bound the ledger size growth.
 
 The ledger rent cost ('Write 1KB' entries in the table) is based on the write fee, rent period and some coefficient. For the temporary storage, the coefficient is 10 months (in ledgers), thus 1 month of temporary entry rent is 1/10 of the rent fee. For persistent storage the coefficient is 1 month (in ledgers) and thus 1 month of rent is equivalent to the write fee.
 
+### Phase 2 (next) {#phase-2}
+
+Phase 2 marks the deployment of a user-ready Mainnet where anyone can deploy and interact with smart contracts. In this phase, the network is ready for more extensive smart contract utilization, so it's the practical beginning of smart contracts on the Stellar network. But even this is not the end of the journey. At the core of smart contracts on the Stellar network is scale and sustainability, and efforts to continue to expand capacity and introduce further capabilities to enhance the network's functionalities will be ongoing. See [this blog post](https://stellar.org/blog/developers/the-stellar-network-s-phased-rollout-of-smart-contracts-the-road-to-mainnet) for more details and share your feedback in [Discord](https://discord.com/invite/zVYdY3ktTn).
+
+| Network setting | Phase 2 Cost (stroops) |
```

Comment:
I don't think we need to separate this by phase, let's just update the values in-place.